src/main/java/com/yelp/nrtsearch/server/luceneserver/field/IntFieldDef.java

```diff
@@ -107,11 +110,40 @@ public Query getRangeQuery(RangeQuery rangeQuery) {
 
   @Override
   public Query getTermQuery(TermQuery termQuery) {
-    return IntPoint.newExactQuery(getName(), termQuery.getIntValue());
+    int intValue;
+    switch (termQuery.getTermTypesCase()) {
+      case TEXTVALUE:
+        intValue = Integer.parseInt(termQuery.getTextValue());
+        break;
+      case INTVALUE:
+        intValue = termQuery.getIntValue();
+        break;
+      default:
+        throw new IllegalArgumentException(
+            "INT field does not support term type: " + termQuery.getTermTypesCase());
+    }
+    return IntPoint.newExactQuery(getName(), intValue);
   }
 
   @Override
   public Query getTermInSetQuery(TermInSetQuery termInSetQuery) {
-    return IntPoint.newSetQuery(getName(), termInSetQuery.getIntTerms().getTermsList());
+    List<Integer> intValues;
+    switch (termInSetQuery.getTermTypesCase()) {
+      case TEXTTERMS:
+        List<Integer> intTermsList = new ArrayList<>();
```

Comment:
nit: initialize the ArrayList to the size of the text term list

Reply:
nit: feel free to ignore.

Are we sure initializing with correct size is more efficient compared to initializing with empty array? I remember reading this a while back: https://shipilev.net/blog/2016/arrays-wisdom-ancients/#_empty_array_instantiation

Not sure if its relevant for Lists or only for primitive arrays..

Reply:
I don't think this applies to ArrayList because of how they grow. Without a capacity, it looks like primitive array allocation is deferred until the first insertion. The size is still dynamic based on the number of elements added, so neither case will have static initialization.